src/AgConfigure.cpp

```diff
@@ -900,11 +864,255 @@ String Configuration::getAbcDayString(int value) {
+    if (unit != "c" && unit != "f") {
+      isInvalid = true;
+    } else {
+      isInvalid = false;
+    }
+  }
+  if (isInvalid) {
+    jconfig[jprop_temperatureUnit] = "c";
+    changed = true;
+    logInfo("toConfig: temperatureUnit changed");
+  }
+
+  /** validate configuration control */
+  if (JSON.typeof_(jprop_configurationControl) != "string") {
+    isInvalid = true;
+  } else {
+    String ctrl = jconfig[jprop_configurationControl];
+    if (ctrl != String(CONFIGURATION_CONTROL_NAME
+                           [ConfigurationControl::ConfigurationControlBoth]) &&
+        ctrl != String(CONFIGURATION_CONTROL_NAME
+                           [ConfigurationControl::ConfigurationControlLocal]) &&
+        ctrl != String(CONFIGURATION_CONTROL_NAME
+                           [ConfigurationControl::ConfigurationControlCloud])) {
+      isInvalid = true;
+    } else {
+      isInvalid = false;
+    }
+  }
+  if (isInvalid) {
+    jconfig[jprop_configurationControl] =
+        String(CONFIGURATION_CONTROL_NAME
+                   [ConfigurationControl::ConfigurationControlBoth]);
+    changed = true;
+    logInfo("toConfig: configurationControl changed");
+  }
+
+  /** Validate post to airgradient cloud */
+  if (JSON.typeof_(jconfig[jprop_postDataToAirGradient]) != "boolean") {
+    isInvalid = true;
+  } else {
+    isInvalid = false;
+  }
+  if (isInvalid) {
+    jconfig[jprop_postDataToAirGradient] = true;
+    changed = true;
+    logInfo("toConfig: postToAirGradient changed");
+  }
+
+  /** validate led bar brightness */
+  if (JSON.typeof_(jconfig[jprop_ledbarBrightness]) != "number") {
+    isInvalid = true;
+  } else {
+    int value = jconfig[jprop_ledbarBrightness];
+    if (value < 0 || value > 100) {
+      isInvalid = true;
+    } else {
+      isInvalid = false;
+    }
+  }
+  if (isInvalid) {
+    jconfig[jprop_ledbarBrightness] = 100;
+    changed = true;
+    logInfo("toConfig: ledBarBrightness changed");
+  }
+
+  /** Validate display brightness */
+  if (JSON.typeof_(jconfig[jprop_displayBrightness]) != "number") {
+    isInvalid = true;
+  } else {
+    int value = jconfig[jprop_displayBrightness];
+    if (value < 0 || value > 100) {
+      isInvalid = true;
+    } else {
+      isInvalid = false;
+    }
+  }
+  if (isInvalid) {
+    jconfig[jprop_displayBrightness] = 100;
+    changed = true;
+    logInfo("toConfig: displayBrightness changed");
+  }
+
+  /** Last OTA */
+  if(JSON.typeof_(jconfig[jprop_lastOta]) != "number") {
+    isInvalid = true;
+  } else {
+    isInvalid = false;
+  }
+  if(isInvalid) {
+    jconfig[jprop_lastOta] = 0;
+    changed = true;
+  }
+
+  if (changed) {
+    saveConfig();
+  }
+}
+
 String Configuration::getFailedMesage(void) { return failedMessage; }
 
 void Configuration::setPostToAirGradient(bool enable) {
-  if (enable != config.postDataToAirGradient) {
-    config.postDataToAirGradient = enable;
+  bool oldEnabled = jconfig[jprop_postDataToAirGradient];
```

Comment:
I don't understand here what "old" means

Reply:
`old` is current configuration valu. It's use to compare to new value.